tests/tests/test_mtls.py

```diff
@@ -356,8 +355,23 @@ def test_mtls_enterprise_without_client_cert(self, setup_ent_mtls):
         # set up the mTLS test environment, without providing client certs
         self.common_test_mtls_enterprise(setup_ent_mtls, algorithm=None, use_hsm=False)
 
-        # wait 30 seconds
-        time.sleep(30)
+        # in here it also may happen, that the client is started earlier, and device registers
+        # as pending. in that case get_devices_status which is called from get_devices will
+        # loop until it runs out of iterations, due to the fact that we expect to have 0 devices.
+        # to prevent this from happening, lets wait a bit, if the device shows as pending,
+        # if it does, decommission it, and then restart the client, and wait to be sure the
+        # device will not re-appear, which is the main idea of the test.
+        device_not_present_timeout_seconds = 30
+        for device in devauth.get_devices_status(
+            "pending", max_wait=device_not_present_timeout_seconds * 0.5, no_assert=True
```

Comment:
Why do you set device_not_present_timeout_seconds to 30 (seconds?) and then multiply it by 0.5?

Reply:
> 

please find the reply below:)